internal/users/localentries/getgrent_c.go

```diff
@@ -7,6 +7,19 @@ package localentries
 #include <stdlib.h>
 #include <pwd.h>
 #include <grp.h>
+
+// Copy a NULL-terminated char** into a new array and return length.
+char **copy_strv(char **strv, int *out_len) {
+    int n = 0;
+    while (strv && strv[n]) n++;
+
+    *out_len = n;
+    char **out = calloc(n + 1, sizeof(char*));
+    for (int i = 0; i < n; i++) {
+        out[i] = strv[i];
+    }
+    return out;
```

Comment:
Since we're returning the length, I'd just call the function strv_to_array or something and would just use `malloc(n * sizeof(char *))` (we don't need an extra pointer and zero init)